### PR TITLE
Add support for P-521 curve with SHA-256/SHA-384 digests

### PR DESCRIPTION
**Pull request by samin-cf**
## Issues:
Address https://github.com/rustls/webpki/issues/267

## Description of changes:
aws-lc-rs has added support for ECDSA P-521 with SHA1, SHA224, SHA256, SHA384 in https://github.com/aws/aws-lc-rs/pull/461.

This PR introduces ECDSA_P521_SHA256 and ECDSA_P521_SHA384 for aws-lc-rs to webpki.

### CODE REVIEW

File: tests/generate.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 Run this script from tests/.  It edits the bottom part of some .rs files and
 drops testcase data into subdirectories as required.
 """
+
```
Comment (djc) [inline tests/generate.py:10]: Maybe revert this unrelated change?

File: tests/generate.py [hunk 2]
```diff
@@ -1126,7 +1135,13 @@ def _revocation_test(
         intermediates_str: str = f"&[{int_a_str}, {int_b_str}]"
 
         def _write_revocation_test(*, owned: bool) -> None:
-            nonlocal crl_paths, expected_error, intermediates_str, test_name, ee_cert_path, root_cert_path
+            nonlocal \
```
Comment (djc) [inline tests/generate.py:1138]: Unrelated.

**Review by djc — APPROVED** [review]
Thanks, LGTM!

File: tests/generate.py [hunk 3]
```diff
@@ -561,6 +562,8 @@ def signatures(force: bool) -> None:
 
     feature_gates = {
         "ECDSA_P521_SHA512": 'all(not(feature = "ring"), feature = "aws_lc_rs")',
+        "ECDSA_P521_SHA256": 'all(not(feature = "ring"), feature = "aws_lc_rs")',
+        "ECDSA_P521_SHA384": 'all(not(feature = "ring"), feature = "aws_lc_rs")',
```
Comment (cpu) [inline tests/generate.py:566]: small nit: order these by digest size (and on L582)

**Review by cpu — APPROVED** [review]
Thanks!

File: tests/generate.py (hunk 2 above)
Comment (cpu) [reply tests/generate.py:1138]: I think this diff is also the cause of the [one failed CI task](https://github.com/rustls/webpki/actions/runs/10025317179/job/27711164274?pr=272)

**Review by djc — APPROVED** [review]

**Comment by djc** [comment]
Used administrative override to squash during merge.
